File: bench/Reference/list_sum.hs
-- list_sum: sum 1..10000 via explicit cons-list build + foldl.
main :: IO ()
main = print (foldl (+) (0 :: Int) (go 10000 []))
  where
    go 0 acc = acc
    go n acc = go (n - 1) (n : acc)

File: bench/Reference/list_sum.erl
#!/usr/bin/env escript
%% sum of 1..10000 built via cons

main(_) ->
    L = build(10000, []),
    io:format("~p~n", [lists:sum(L)]).

build(0, Acc) -> Acc;
build(N, Acc) -> build(N - 1, [N | Acc]).
